AxeWP/wp-graphql-gravity-forms
# fix: installation errors when upgrading plugin via the WordPress backend

## What


This PR updates the `yahnis-elsts/plugin-update-checker ` dependency to its latest version, and fixes a couple issue that were causing updating the plugin via WP's backend to fail.

## Why

Closes #386 


## How

- The immediate issue reported in #386 is caused by a change to GitHub's API which was causing PUC to try and install the generated `.graphql` artifact. This is fixed by updating the plugin (and for extra measure, setting the zip filename explicitly.

- With that solved however, PUC instead began throwing issues due to the fact that the release zip stored its assets top-level, instead of in `wp-graphql-gravity-forms/*`. This has been fixed in the GitHub workflow.

## Testing Instructions


** ⚠️ This will replace the plugin folder in your test environment with the one from the fictitious release zip!!! ⚠️**
1. Add this filter to detect the fictious future release:

```php
add_filter( 'graphql_gf_update_repo_url', fn() => 'https://github.com/justlevine/wp-graphql-gravity-forms' );
```
2. See the available update:
![image](https://github.com/AxeWP/wp-graphql-gravity-forms/assets/29322304/fe51744b-315a-4de6-8182-984ad06963f9)

3. Update 🤞
![image](https://github.com/AxeWP/wp-graphql-gravity-forms/assets/29322304/6a4a295e-f345-4148-810c-dbcd3519f7c0)


## Additional Info


## Checklist:


- [x] This PR is tested to the best of my abilities.
- [x] This PR follows the WordPress Coding Standards. 
- [x] This PR has proper inline documentation. 
- [x] This PR has unit tests to verify the code works as intended.
- [x] The changes in this PR have been noted in CHANGELOG.md
